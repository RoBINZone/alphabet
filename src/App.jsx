import React, { useState } from "react";
import { WelcomeScreen } from "./components/WelcomeScreen";
import { GameplayScreen } from "./components/GameplayScreen";
import "./App.css";
import { StatisticsScreen } from "./components/StatisticsScreen";
import { getDateTime } from "./services/statistic-service";

function App() {
  const [screen, setScreen] = useState("WELCOME");
  const [startTime, setStartTime] = useState(null);

  return (
    <>
      {screen === "WELCOME" && (
        <WelcomeScreen
          onStart={() => {
            setStartTime(getDateTime());
            setScreen("GAMEPLAY");
          }}
        />
      )}
      {screen === "GAMEPLAY" && (
        <GameplayScreen startTime={startTime} onScreenChange={setScreen} />
      )}
      {screen === "STATISTICS" && (
        <StatisticsScreen onScreenChange={setScreen} />
      )}
    </>
  );
}

export default App;
