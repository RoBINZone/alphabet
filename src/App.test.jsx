import React from "react";
import { render } from '@testing-library/react';
import { expect, it, describe } from 'vitest';
import App from './App';

describe('App', () => {
  it('should match snapshot', () => {
    const { asFragment } = render(<App />);
    expect(asFragment()).toMatchSnapshot();
  });
});

