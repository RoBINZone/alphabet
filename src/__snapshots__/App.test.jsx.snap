// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`App > should match snapshot 1`] = `
<DocumentFragment>
  <div>
    <p>
      Вітаємо в грі "Алфавіт". Правила прості: вибирайте з двох літер ту, що йде першою в алфавіті.
    </p>
    <button>
      Почати гру
    </button>
  </div>
</DocumentFragment>
`;
